clients/application-amqp/src/test/java/org/eclipse/hono/application/client/amqp/ProtonBasedCommandSenderTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.qpid.proton.amqp.messaging.Accepted;
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.HonoConnection;
+import org.eclipse.hono.client.amqp.test.AmqpClientUnitTestHelper;
+import org.eclipse.hono.test.VertxMockSupport;
+import org.eclipse.hono.util.MessageHelper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.EventBus;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.proton.ProtonDelivery;
+import io.vertx.proton.ProtonMessageHandler;
+import io.vertx.proton.ProtonQoS;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * Tests verifying behavior of {@link ProtonBasedCommandSender}.
+ *
+ */
+@ExtendWith(VertxExtension.class)
+@Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+public class ProtonBasedCommandSenderTest {
+
+    private final ArgumentCaptor<Handler<ProtonDelivery>> dispositionHandlerCaptor = VertxMockSupport.argumentCaptorHandler();
+    private final ArgumentCaptor<Message> messageCaptor = ArgumentCaptor.forClass(Message.class);
+    private ProtonBasedApplicationClient client;
+    private ProtonSender sender;
+    private ProtonDelivery protonDelivery;
+
+    /**
+     * Sets up the fixture.
+     */
+    @BeforeEach
+    void setUp() {
+        final var vertx = mock(Vertx.class);
+        when(vertx.eventBus()).thenReturn(mock(EventBus.class));
+        final HonoConnection connection = AmqpClientUnitTestHelper.mockHonoConnection(vertx);
+
+        final ProtonReceiver receiver = AmqpClientUnitTestHelper.mockProtonReceiver();
+        when(connection.createReceiver(
+                anyString(),
+                any(ProtonQoS.class),
+                any(ProtonMessageHandler.class),
+                anyInt(),
+                anyBoolean(),
+                VertxMockSupport.anyHandler())).thenReturn(Future.succeededFuture(receiver));
+
+        protonDelivery = mock(ProtonDelivery.class);
+        when(protonDelivery.remotelySettled()).thenReturn(true);
+        when(protonDelivery.getRemoteState()).thenReturn(new Accepted());
+
+        sender = AmqpClientUnitTestHelper.mockProtonSender();
+        when(sender.send(any(Message.class), VertxMockSupport.anyHandler())).thenReturn(protonDelivery);
+        when(connection.createSender(anyString(), any(), any())).thenReturn(Future.succeededFuture(sender));
+
+        client = new ProtonBasedApplicationClient(connection);
```

Comment:
this should be a new *ProtonBasedCommandSender*, shouldn't it?

Reply:
My bad, I just moved those tests to the new class and missed to replace it with `ProtonBasedCommandSender`. Thanks for pointing out and I fixed it now.

Reply:
No worries, that's what the review is there for ;-)